forest/daemon/src/daemon.rs

```diff
@@ -538,37 +541,48 @@ async fn prompt_snapshot_or_die(
 async fn sync_from_snapshot<DB>(
     config: &Config,
     state_manager: &Arc<StateManager<DB>>,
+    proof_params_required: bool,
 ) -> Result<(), anyhow::Error>
 where
     DB: Store + Send + Clone + Sync + Blockstore + 'static,
 {
     if let Some(path) = &config.client.snapshot_path {
         let stopwatch = time::Instant::now();
+
         let validate_height = if config.client.snapshot {
             config.client.snapshot_height
         } else {
             Some(0)
         };
 
-        match import_chain::<_>(
+        let ensure_params_downloaded_task_handle = if proof_params_required {
+            Some(tokio::spawn(async {
+                // Add a small delay to avoid import chain progress bar being interrupted.
+                tokio::time::sleep(Duration::from_secs(1)).await;
```

Comment:
This is a bit brittle approach, but let's investigate how to sync this properly in another issue, shall we?